policies/testing.md

```diff
@@ -0,0 +1,42 @@
+Testing Policy
+==============
+
+This applies to all stable and development branches of the main code repository.
+Except where noted below:
+
+- All Pull Requests adding new functionality to the applications, libraries,
+providers or engines must include suitable tests.
+- All Pull Requests fixing a functional behavior defect in the applications,
+libraries, providers or engines must include a test for that defect.
+
+Pull Requests that do not change the functional behaviour of the applications,
+libraries, providers or engines do not require tests to be added. For example
+the following types of changes do not require tests:
+ - Changes to comments, formatting, internal naming or similar
+ - Fixes for performance
```

Comment:
If it's about performance, and it was not something we monitored under the performance testing before, should we not add it to that? Clearly someone thought it was important enough to fix it so that we want to monitor it for regressions.

Reply:
I've added a non-mandatory "Recommendations" section and added recommendation to consider whether a performance test is appropriate.